common/src/test/java/com/sequenceiq/cloudbreak/util/PublicKeyReaderUtilTest.java

```diff
@@ -1,21 +1,208 @@
 package com.sequenceiq.cloudbreak.util;
 
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.fail;
+
 import java.security.PublicKey;
+import java.security.interfaces.EdECPublicKey;
+import java.security.interfaces.RSAPublicKey;
+import java.security.spec.EdECPoint;
 
-import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 
+import com.sequenceiq.cloudbreak.util.PublicKeyReaderUtil.PublicKeyParseException;
+
 class PublicKeyReaderUtilTest {
 
     @Test
-    public void testRsaKeyMustUseAlgorithmRSA() throws PublicKeyReaderUtil.PublicKeyParseException {
+    void loadTestRsaKeyMustUseAlgorithmRSAWhenOpenSSH() throws PublicKeyParseException {
         PublicKey rsaKey = PublicKeyReaderUtil.load("ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQDfz7EED7gE8qbDKg6tUMtL3yW/" +
                 "W1kiGzWBbVhOMiFYjnVje8mNF5jNe8rj+tg6fKA9dMpULrTIaFC5Dbhq4c0vCRqTbzrPkrXCURAdWx1lb+NzfTIOqLVeUZlnh" +
                 "NPXBPVss+pnkawsI4HUiMRuScQnzrbvY9HcLhb69Qe9sZvSx1R46F9BEOZduX3QJxV2NBer6zJAZxv/YJt/ObzSFSnWN5isG+X" +
                 "x/tIUZHighuqH7k0pzanp8IH11O6qXuCyqk9c7110w/vlQ+N8N3UhtAOhPCljKjYJo0KgF9C//l9eVKi/QICGVbr6vzvAyiWg" +
                 "uf1lMHUEmsDlGQqfq/nnKCQYlz2V6oOaPt6ggmdmNCppFOSXri0dm1X4TnLASpIFetdjFGTU+sH75SmSrPlYn5HOxcqeVGgHNIZ2" +
                 "taY+u12DWtY7cqFXvHE6pFmPme2Z0wQrAq/rH2BqGl3pkWVDIItbtsVsVTlS6E1Y/dXJkjaBiBZTc7aeBLa58oj5F4JYb" +
-                "sc= test@test.local");
-        Assertions.assertTrue(rsaKey.getAlgorithm().equals("RSA"));
+                "sc= test@test.local", false);
+        assertEquals("RSA", rsaKey.getAlgorithm());
+        if (rsaKey instanceof RSAPublicKey rsaPublicKey) {
+            assertThat(rsaPublicKey.getPublicExponent().toString()).isEqualTo("65537");
+            assertThat(rsaPublicKey.getModulus().toString()).isEqualTo(
+                    "15716563600099515211328423611954225824916533615230691376921478986333060601" +
+                    "57552226919808527872386330392237985485200227383711345332616167018657122816541835915" +
+                    "10152175666741773856946951958086291801538991290666845716385778192871160342792145141" +
+                    "097809456369701048006656617692262816450908563750970503449532801209031"
```

Comment:
Hint: Might use """ textblock quote

Reply:
It is a single line without linebreaks. :smile: